src/pybind/mgr/dashboard/frontend/src/app/ceph/smb/smb.model.ts

```diff
@@ -0,0 +1,14 @@
+export interface SMBCluster {
+  cluster_id: string;
+  auth_mode: string;
+  intent: string;
+  domain_settings: DomainSettings;
+  user_group_settings: string[];
```

Comment:
I think @Pegonzal mentioned these needs to be optional fields as they can be either domain_settings or user_group_settings
```suggestion
  domain_settings?: DomainSettings;
  user_group_settings?: string[];
```